core/src/main/java/org/javarosa/core/model/instance/TreeElement.java

```diff
@@ -648,6 +694,13 @@ public void setAttribute(String namespace, String name, String value) {
         if (attributes == null) {
             this.attributes = new Vector<TreeElement>();
         }
+
+        if ("".equals(namespace)) {
+            // normalize to match how non-attribute TreeElements store namespaces
+            // NOTE PLM: "" and null are quite conflated, especially in read/writeExternal.
+            namespace = null;
+        }
```

Comment:
^ worth double checking.
There is some crazy stuff going on with `emptyIfNull` and `nullIfEmpty` in the read/writeExternal methods that messes things up, so that is why this piece of code is here.